### PR TITLE
Add roomId as new property for user entity

### PR DESCRIPTION
**Pull request by isisblack**
- Create doPut method in UserServlet
      - takes current roomId from url parameter
      - sets roomId as new property in user entity
demo:  https://screencast.googleplex.com/cast/NjUyMjU0MjM0Njk5MzY2NHwzMjM4NTA3OC0wNw
TODO: DRY with getUserId() (in Salil's "Refactor getUserId and readJSONRequest functions" PR)

### CODE REVIEW

File: src/main/java/com/google/songgame/servlets/UserServlet.java [hunk 1]
```diff
@@ -49,9 +53,43 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
   }
 
+  @Override
+  public void doPut(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Map<String, String> roomProperties = readJSONFromRequest(request);
+
+    Query userQuery =
+        new Query("User").addFilter("userId", FilterOperator.EQUAL, roomProperties.get("userId"));
+    PreparedQuery result = datastore.prepare(userQuery);
+    Entity currentUser = result.asSingleEntity();
```
Comment (salnad) [inline src/main/java/com/google/songgame/servlets/UserServlet.java:63]: Currently the room seems to store only one user. Is the userId only for admins? And other users aren't stored in the rooms? If so can you change the property into something more generic, like adminUserId?

Comment (isisblack) [reply src/main/java/com/google/songgame/servlets/UserServlet.java:63]: @salnad The user is stored in the room, but the current room is stored for each user one at a time as they join the room

Comment (salnad) [reply src/main/java/com/google/songgame/servlets/UserServlet.java:63]: Ah alright, I think I was misunderstanding this. Thought it was RoomServlet not UserServlet. Looks good!

**Comment by pruess-google** [comment]
> I might not be fully understanding the flow here.
> 
> The code to put the room id in the user is being called from "loadUsernames" function which to me isn't obvious that it should do that.
> 
> Since we already have code to add a user id to a room, would it make sense to put the reciprocal of that in the same place? (i.e. adding the room id to the user.).
> 
> It could go somewhere around here:
> https://github.com/googleinterns/step105-2020/blob/master/src/main/java/com/google/songgame/servlets/RoomServlet.java#L67

I agree with Aleksey's suggestion. The best and easiest way to make this change would be to add the roomId to the User entity at the same time we add the userId to the Room entity.

I'm curious, what do you need the room ID on the user entity?

**Pull request closed without merging**